java/gazelle/private/servermanager/embedded.go

```diff
@@ -0,0 +1,82 @@
+package servermanager
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+// The embedded javaparser server (and runner script) will be materialized at runtime every time we run gazelle.
+// This is not a problem for now, as the javaparser server is tiny.
+// However, we may want to get clever and materialize to a more permanent location in the future if the server becomes significant.
+//
+//go:embed javaparser.jar
+var javaparserDeployJar []byte
+
+func materializeJavaparser(tmpdir string) (string, error) {
+	jarPath := filepath.Join(tmpdir, "javaparser.jar")
+	err := os.WriteFile(jarPath, javaparserDeployJar, 0644)
+	if err != nil {
+		return "", err
+	}
+	return jarPath, nil
+}
+
+const unixRunerTemplate = `#!/bin/sh
+%s %s -jar %s ${@}
+`
+
+const windowsRunnerTemplate = `
+@echo off
+setlocal
+
+"%s" %s -jar "%s" %*
+
+endlocal
+`
+
+func createRunner(javaparserLocation string, jvmFlags []string, tmpdir string) (string, error) {
+	runnerPath := filepath.Join(tmpdir, "runjavaparser")
+	runnerTemplate := unixRunerTemplate
+
+	if runtime.GOOS == "windows" {
+		runnerPath += ".exe"
+		runnerTemplate = windowsRunnerTemplate
+	}
+
+	javaBin :=
+		os.Getenv("GAZELLE_JAVA_BIN")
```

Comment:
Should this GAZELLE_JAVA_BIN be:

1. a const so it carries the doc string?
2. renamed to a more verbose yet systemic GAZELLE_JAVA_JAVAHOME

Reply:
Yep! Done :D Great idea.